src/index.ts

```diff
@@ -44,12 +56,47 @@ const createImportFinallyResolve = (qiankunName: string) => {
 
 export type MicroOption = {
   useDevMode: boolean
+  rules?: AssetsRule[]
 }
 type PluginFn = (qiankunName: string, microOption: MicroOption) => PluginOption;
 
 const htmlPlugin: PluginFn = (qiankunName, microOption) => {
   let isProduction: boolean
 
+  const {
+    rules = DefaultRules
+  } = microOption
+
+  let config: ResolvedConfig
+
+  async function transform (code: string, id: string) {
+    const s = new MagicString(code)
+
+    let match
+
+    for (const rule of rules) {
+      rule.match.lastIndex = 0
+      // eslint-disable-next-line no-cond-assign
+      while ((match = rule.match.exec(code))) {
+        const start = code.indexOf(match[1])
+        const end = start + match[1].length
+        const url = match[1]
+        if (!url || !microOption.useDevMode) { continue }
+        const appendBase = `http://127.0.0.1:${config.server.port}/`
+        const emptyUrl = url.startsWith('/') ? url.substring(1) : url
```

Comment:
This will cause problems when
url = '//www.baidu.com/img/PCtm_d9c8750bed0b3c7d089fa7d55720d6cf.png' 
or
url = 'https://www.baidu.com/img/PCtm_d9c8750bed0b3c7d089fa7d55720d6cf.png'